src/main/java/net/floodlightcontroller/sos/SOS.java

```diff
@@ -264,7 +276,66 @@ public boolean isCallbackOrderingPostreq(OFType type, String name) {
 			return false;
 		}
 	}
+	/**
+	 * @author Khayam Anjam kanjam@clemson.edu
+	 * Now Instead of sending a UDP packet. We will make a call to Agent's rest server
+	 */
+	private boolean sendInfoToAgent(SOSConnection conn, boolean isClientSideAgent) {
+		HttpClient httpClient = new DefaultHttpClient();
+		JSONObject requestObject = new JSONObject();
+		try {
+			requestObject.put("is-client-agent", isClientSideAgent)
+					.put("transfer-id", conn.getTransferID().toString())
+					.put("client-ip", conn.getClient().getIPAddr().toString())
+					.put("client-port", conn.getClient().getTcpPort().toString())
+					.put("server-agent-ip", conn.getServerSideAgent().getIPAddr().toString())
+					.put("num-parallel-socks", conn.getNumParallelSockets())
+					.put("buffer-size", conn.getBufferSize())
+					.put("queue-capacity", conn.getQueueCapacity())
+					.put("server-ip", conn.getServer().getIPAddr().toString())
+					.put("server-port", conn.getServer().getTcpPort().toString()) ;
+
+		StringBuilder uriBuilder = new StringBuilder(3);
+		uriBuilder.append(SOSAgentUtils.HTTP_PRESTRING);
+		if (isClientSideAgent) {
+			uriBuilder.append(SOSAgentUtils.addressBuilder(conn.getClientSideAgent().getRestIpAddr().toString(),
+					conn.getClientSideAgent().getRestPort().toString()));
+			log.info("Chosen Client side agent with IP {}", conn.getClientSideAgent().getIPAddr().toString());
+		}
+		else {
+			uriBuilder.append(SOSAgentUtils.addressBuilder(conn.getServerSideAgent().getRestIpAddr().toString(),
+					conn.getServerSideAgent().getRestPort().toString()));
+			log.info("Chosen Server side agent with IP {}", conn.getServerSideAgent().getIPAddr().toString());
+		}
+		uriBuilder.append(SOSAgentUtils.PathBuilder(SOSAgentUtils.REQUEST_PATH));
+
+		HttpPost httRequest = new HttpPost(uriBuilder.toString());
```

Comment:
Not really an issue but I think this is a typo

Reply:
Fixed.